src/main/java/reposense/parser/ArgsParser.java

```diff
@@ -274,30 +273,67 @@ public static CliArguments parse(String[] args) throws HelpScreenException, Pars
                 }
             }
 
-            boolean isSinceDateProvided = cliSinceDate.isPresent();
-            boolean isUntilDateProvided = cliUntilDate.isPresent();
-            boolean isPeriodProvided = cliPeriod.isPresent();
-            if (isSinceDateProvided && isUntilDateProvided && isPeriodProvided) {
+            // CLI arguments are treated with the highest priority and will throw an Exception if a conflict occurs
+            if (cliSinceDate.isPresent() && cliUntilDate.isPresent() && cliPeriod.isPresent()) {
                 throw new ParseException(MESSAGE_HAVE_SINCE_DATE_UNTIL_DATE_AND_PERIOD);
             }
             int numCloningThreads = results.get(CLONING_THREADS_FLAG[0]);
             int numAnalysisThreads = results.get(ANALYSIS_THREADS_FLAG[0]);
 
+            boolean isSinceDateProvided = cliSinceDate.isPresent() || reportConfig.getSinceDate().isPresent();
+            boolean isUntilDateProvided = cliUntilDate.isPresent() || reportConfig.getUntilDate().isPresent();
+            boolean isPeriodProvided = cliPeriod.isPresent() || reportConfig.getPeriod().isPresent();
+
+            // Report config is treated with less priority and will be ignored if there is a conflict
+            if (isSinceDateProvided && isUntilDateProvided && isPeriodProvided) {
+                logger.warning(MESSAGE_IGNORING_REPORT_SINCE_DATE_UNTIL_DATE_AND_PERIOD);
+                reportConfig.ignoreSinceUntilDateAndPeriod();
+                isSinceDateProvided = cliSinceDate.isPresent();
+                isUntilDateProvided = cliUntilDate.isPresent();
+                isPeriodProvided = cliPeriod.isPresent();
+            }
+            if (cliSinceDate.isPresent() && reportConfig.getSinceDate().isPresent()) {
+                logger.warning(MESSAGE_IGNORING_REPORT_CONFIG_SINCE_DATE);
+            }
+            if (cliUntilDate.isPresent() && reportConfig.getUntilDate().isPresent()) {
+                logger.warning(MESSAGE_IGNORING_REPORT_CONFIG_UNTIL_DATE);
+            }
+            if (cliPeriod.isPresent() && reportConfig.getPeriod().isPresent()) {
+                logger.warning(MESSAGE_IGNORING_REPORT_CONFIG_PERIOD);
+            }
+
+            // Set the values to those provided by the CLI arguments first, otherwise take from report config
+            if (cliPeriod.isPresent()) {
+                tempPeriod = cliPeriod;
+            } else if (reportConfig.getPeriod().isPresent()) {
+                tempPeriod = reportConfig.getPeriod();
+            }
+            if (cliSinceDate.isPresent()) {
+                tempSinceDate = cliSinceDate;
+            } else if (reportConfig.getSinceDate().isPresent()) {
+                tempSinceDate = reportConfig.getSinceDate();
+            }
+            if (cliUntilDate.isPresent()) {
+                tempUntilDate = cliUntilDate;
+            } else if (reportConfig.getUntilDate().isPresent()) {
+                tempUntilDate = reportConfig.getUntilDate();
+            }
+
```

Comment:
Find the whole logic above a bit cumbersome. There is a better way to avoid a few of the checks by writing the if conditions in a clever manner. I will leave that to you to figure out.

Reply:
I thought of the following, not sure if that is what you are looking for?

```java
tempPeriod = Optional.ofNullable(cliPeriod.orElse(reportConfig.getPeriod().orElse(null)));
tempSinceDate = Optional.ofNullable(cliSinceDate.orElse(reportConfig.getSinceDate().orElse(null)));
tempUntilDate = Optional.ofNullable(cliUntilDate.orElse(reportConfig.getUntilDate().orElse(null)));
```

Reply:
I meant for the whole page in general. Not just the last three lines. Cause I can see you see checking if arguments are present or not 2/3 times. So I didn't like that part of the logic. Can try to make sure that you check these only once and make it work.

Reply:
Sorry @Tejas2805 I still can't figure out how to better write this part of the code. My train of thought for this is:

1. Check if all 3 arguments are provided (either from report config, from CLI, or from both). If yes, we ignore all of them and take only the CLI arguments.
2. For each of the 3 arguments, if it is given in both the CLI and in the report config, we give the user a warning.
3. Afterwards, we take only the CLI arguments.
4. We then process the actual since and until dates, based on the processed information. If both since and until dates are provided (which only happens in CLI), the period is ignored without warning.

I feel that they are all checking different types of conditions, so I can't wrap my head around how to better craft this section.